platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java

```diff
@@ -414,10 +482,131 @@ private boolean onItemLongClick(int position) {
     protected void onActivityResult(int requestCode, int resultCode,
                                     Intent data) {
         Log.d(TAG, "onActivityResult ");
+        Log.d(TAG, "requestCode: " + resultCode);
         Log.d(TAG, "resultCode: " + resultCode);
 
-        // Close recursively the activity stack
-        if (resultCode == Activity.RESULT_OK) {
+        if (requestCode == R.id.import_dataset &&
+            resultCode == Activity.RESULT_OK) {
+            Log.d(TAG, "handling import dataset(s)");
+            File externalFilesDir = getExternalFilesDir(null);
+            if (externalFilesDir == null || data == null) {
+                return;
+            }
+
+            String importDatasetPath =
+                externalFilesDir.getAbsolutePath() + "/Imported Datasets/";
+            new File(importDatasetPath).mkdir();
+            Context context = getApplication().getApplicationContext();
+            ContentResolver resolver = getContentResolver();
+
+            if (data.getClipData() != null) {
+                for (int i = 0; i < data.getClipData().getItemCount(); i++) {
+                    Uri uri = data.getClipData().getItemAt(i).getUri();
+                    DocumentFile documentFile =
+                        DocumentFile.fromSingleUri(context, uri);
+                    String importFilePath =
+                        importDatasetPath + documentFile.getName();
+                    try {
+                        InputStream input = resolver.openInputStream(uri);
+                        QFieldUtils.inputStreamToFile(input, importFilePath);
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                }
+            } else {
+                Uri uri = data.getData();
+                DocumentFile documentFile =
+                    DocumentFile.fromSingleUri(context, uri);
+                String importFilePath =
+                    importDatasetPath + documentFile.getName();
+                try {
+                    InputStream input = resolver.openInputStream(uri);
+                    QFieldUtils.inputStreamToFile(input, importFilePath);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+
+            Intent intent = new Intent(this, QFieldProjectActivity.class);
+            intent.putExtra("path", importDatasetPath);
+            intent.putExtra("label",
+                            getString(R.string.favorites_imported_datasets));
+            startActivityForResult(intent, 123);
```

Comment:
Magic number?

Reply:
@m-kuhn , yeah, the number we've been using forever there when opening project view/folder activity